Sources/DarkModeCore/DMTraitCollection.m

```diff
@@ -175,19 +185,55 @@ - (instancetype)init {
   return self;
 }
 
-+ (void)registerApplication:(UIApplication *)application {
-  _registeredApplication = application;
+// MARK: - Observer Registration
++ (void)registerWithApplication:(UIApplication *)application syncImmediately:(BOOL)syncImmediately animated:(BOOL)animated {
+  __weak UIApplication *weakApp = application;
+  _userInterfaceStyleChangeHandler = ^(DMTraitCollection *traitCollection, BOOL animated) {
```

Comment:
Would this block be called when there is a new window opened? Probably we should observe `UIScene. willConnectNotification` or a more appropriate notification, and synchronize interface style.

Reply:
Let’s add this to the checklist for now.

Reply:
This may break apps with iPadOS multiple windows support. Don’t forget to look into this issue before release.